src/ride_type.cpp

```diff
@@ -221,6 +221,11 @@ const RideType *RideInstance::GetRideType() const
  * Immediately remove all guests and staff which is inside the ride.
  */
 
+/**
+ * \fn void RideInstance::GetAllPiecePositions()
+ * Returns a vector with the coordinates of all voxels which are occupied by this ride.
+ */
```

Comment:
Your documentation is at the wrong abstraction level.

You are describing what the code is quite literally doing. That isn't much help, since you can read code to figure that out.
What you should write instead is the higher level intent of the function.
Perhaps something like "collect all voxels used by the ride".

vectors and coordinates are the irrelevant low level details, the shape of how you represent a collection of voxels here.

You may not realize it, but in this program, documentation is leading above code. The documentation described the high level intent of the function, the code realizes that intent in some form. If documentation and code are in conflict, intent wins.

The reason is that you read the documentation when performing a function call. You don't go in reading 150 lines of dense code to figure out how to call the thing. In other words, all calls to a function are based on the written intent and not on what the code is actually doing. Making code follow intent makes code fixes more local.